### PR TITLE
Update dependency com.android.tools.build:gradle to v7.4.0

### PR DESCRIPTION
**Pull request by 3flex**

### CODE REVIEW

**Comment by 3flex** [comment]
`Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8`

Now we could possibly jump to Java 11 minimum to execute the Gradle plugin, but AGP 8.0 bumps again to Java 17 which will be too aggressive, so we should probably solve for this now and maintain compatibility with Java 8. Hopefully that's feasible because this is a compile only dependency, so we can compile with Java 11 dependencies but produce Java 8 bytecode.

**Comment by chao2zhang** [comment]
We probably need to validate if Android projects can still apply detekt-gradle-plugin if there is internal checks inside AGP. I am referring to https://developer.android.com/studio/releases/gradle-plugin#jvm-11-bytecode
```
This means that if you compile against AGP, or write custom Lint checks, you need to start targeting JVM 11 bytecode
```

**Comment by BraisGabin** [comment]
I don't get how this work. You are not adding the agp to the jdk8 run. Shouldn't those tests fail with class not found or something similar?

**Comment by 3flex** [comment]
> This means that if you compile against AGP, or write custom Lint checks, you need to start targeting JVM 11 bytecode

We shouldn't have any problems here, since we're still compiling against AGP 7.0 which doesn't have that limitation, and we're not writing custom lint checks.

**Comment by 3flex** [comment]
> Shouldn't those tests fail with class not found or something similar?

Android tests are already skipped if JVM is lower than 11.

https://github.com/detekt/detekt/blob/0245b0f24b032649908efa37f78b6abd2374efee/detekt-gradle-plugin/src/functionalTest/kotlin/io/gitlab/arturbosch/detekt/DetektAndroidSpec.kt#L13-L15

**Review by BraisGabin — APPROVED** [review]
I really like this addition to compile against min versión but tests against last know version :)

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -22,7 +22,13 @@ kotlin-reflection = { module = "org.jetbrains.kotlin:kotlin-reflect", version.re
...
 kotlinx-coroutines = "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.4"
 
-android-gradle = "com.android.tools.build:gradle:7.3.1"
+# This represents the the oldest AGP version that is supported by detekt.
+# This should only be updated when updating the minimum version supported by detekt's Gradle plugin.
+android-gradle-minSupported = "com.android.tools.build:gradle:7.0.0"
+
+# This version of AGP is used for testing and should be updated when new AGP versions are released to ensure detekt's
+# Gradle plugin remains compatible.
+android-gradle-maxSupported = "com.android.tools.build:gradle:7.4.0"
```
[the remark this answers is not in the document]
Comment (chao2zhang) [reply gradle/libs.versions.toml:31]: ❤️

**Review by chao2zhang — APPROVED** [review]

**Pull request merged**